crates/bevy_ecs/src/system/query.rs

```diff
@@ -10,235 +10,265 @@ use crate::{
 };
 use std::{any::TypeId, borrow::Borrow, fmt::Debug};
 
-/// Provides scoped access to components in a [`World`].
+/// [System parameter] that provides selective access to data stored in a [`World`].
```

Comment:
```suggestion
/// [System parameter] that provides selective access to the [`Component`] data stored in a [`World`].
```

Reply:
It doesn't provide access to resources, and this is important to highlight up front.